src/main/java/org/folio/rest/impl/PiecesAPI.java

```diff
@@ -51,6 +109,32 @@ public void deleteOrdersStoragePiecesById(String id, Map<String, String> okapiHe
   @Validate
   public void putOrdersStoragePiecesById(String id, Piece entity, Map<String, String> okapiHeaders,
       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    PgUtil.put(PIECES_TABLE, entity, id, okapiHeaders, vertxContext, PutOrdersStoragePiecesByIdResponse.class, asyncResultHandler);
+    pgClient.withTrans(conn -> updatePiece(conn, entity, id)
+      .compose(ignore -> auditOutboxService.savePieceOutboxLog(conn, entity, PieceAuditEvent.Action.EDIT, okapiHeaders)))
+      .onComplete(ar -> {
+        if (ar.succeeded()) {
+          log.info("Update piece complete, id={}", id);
+          auditOutboxService.processOutboxEventLogs(okapiHeaders);
+          asyncResultHandler.handle(buildNoContentResponse());
+        } else {
+          log.error("Update piece failed, id={}", id, ar.cause());
+          asyncResultHandler.handle(buildErrorResponse(ar.cause()));
+        }
+      });
   }
+
+  private Future<RowSet<Row>> updatePiece(Conn conn, Piece piece, String id) {
+    log.debug("Updating piece with id={}", id);
+
+    return conn.update(TableNames.PIECES_TABLE, piece, id)
+      .compose(DbUtils::failOnNoUpdateOrDelete)
+      .onSuccess(rowSet -> log.info("Piece successfully updated, id={}", id))
+      .onFailure(e -> log.error("Update piece failed, id={}", id, e));
+  }
+
+  @Override
+  protected String getEndpoint(Object entity) {
+    return null;
```

Comment:
Please implement it as in other modules to have standartized approach.
And after this you can change postOrdersStoragePieces() to use this method
String endpoint = HelperUtils.getEndpoint(OrdersStoragePieces.class) + entity.getId();